src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -57,49 +64,48 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, string indexDir
 
-        public async Task<bool> PushToFeedFlat(IEnumerable<string> items, bool allowOverwrite = false)
+        public async Task<bool> PushItemsToFeedAsync(IEnumerable<string> items, bool allowOverwrite)
         {
-            if (CancellationToken.IsCancellationRequested)
-            {
-                Log.LogError("Task PushToFeedFlat cancelled");
-                CancellationToken.ThrowIfCancellationRequested();
-            }
-
-            using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
-            {
-                await Task.WhenAll(items.Select(item => PushItemToFeed(item, feed.RelativePath, clientThrottle, allowOverwrite, true)));
-            }
-            return !Log.HasLoggedErrors;
-        }
+            Log.LogMessage(MessageImportance.Low, $"START pushing items to feed");
 
-        public async Task<bool> PushItemToFeed(string item, string relativePath, SemaphoreSlim clientThrottle, bool allowOverwrite, bool isFlat)
-        {
             try
             {
-                string uploadPath = feed.CalculateBlobPath(item, relativePath);
-                string packageDirectory = feed.CalculateRelativeUploadPath(item, relativePath);
+                
 
-                await UploadAsync(CancellationToken, item, uploadPath, clientThrottle, allowOverwrite);
+                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(source.ConnectionString);
+                AzureFileSystem fileSystem = new AzureFileSystem(new LocalCache(), new Uri(source.Path), new Uri(source.Path), storageAccount, source.Name, source.FeedSubPath);
 
-                if (!isFlat)
+                // In case the first Push attempt fails with an InvalidOperationException we Init the feed and retry the Push command once.
+                // Sleet internally retries 5 times on each package when the push operation fails so we don't need to retry ourselves.
+                try
+                {
+                    bool result = await PushAsync(items.ToList(), allowOverwrite);
+                    return result;
+                }
+                catch (InvalidOperationException ex) when (ex.Message.Contains("init"))
                 {
-                    List<string> listAzureBlobs = await ListAzureBlobs.ListBlobs(Log, feed.AccountName, feed.AccountKey, feed.ContainerName, packageDirectory);
-                    if (!listAzureBlobs.Any(x => x.Contains(uploadPath)))
+                    Log.LogWarning($"Sub-feed {source.FeedSubPath} has not been initialized. Initializing now...");
+                    bool result = await InitAsync();
+
+                    if (result)
                     {
-                        throw new Exception($"Uploaded package {uploadPath} is not present on feed. Cannot update index.json.");
+                        Log.LogMessage($"Initializing sub-feed {source.FeedSubPath} succeeded!");
+                        result = await PushAsync(items.ToList(), allowOverwrite);
```

Comment:
Don't do this in the catch block. Use a retry pattern.

Reply:
Is this a general style, or are there issues that come up? (Is it because awaiting in catch is a recent C# feature?) I suggested in my review to move away from the `for (int i = 0; i <= 1; i++)` because it was raising some questions.

Reply:
In general you shouldn't be calling code that might throw inside a catch block as if it throws there it will terminate the process.

Reply:
As for the for loop retry pattern I agree it is a little odd but not a big deal. I especially find the `<=1` part odd instead of just a `<2`.